### ISSUE TITLE
GetFinalPathNameByHandleW crashes with error messages when trying to run Firefox 61

### ISSUE THREAD

**Issue by Svyatpro** [MyTDT-Mysoft/DllCompat#7]
This is very unusual use, but I forward some functions inside One-Core-API's kernel32 to kernel3x.dll.
When forwarding GetFinalPathNameByHandleW to kernel3x.dll I get two error messages when loading Firefox 61 on Windows Server 2003 SP2 RUS. First I get "Yay!" and the second one "Filesize=0..."

![image](https://user-images.githubusercontent.com/3678456/46385696-261c1900-c6c6-11e8-9f9e-5ebc0479e79d.png)
![image](https://user-images.githubusercontent.com/3678456/46385719-464bd800-c6c6-11e8-95bf-1bf293fc9330.png)

**Comment by Svyatpro** [comment]
I don't really know how you can reproduce it, I can share some OCAPI's dlls with all forwards.

**Comment by MyTDT-Mysoft** [comment]
yeah they are "important" debug messages... the "yay" is because of the
method that it expects to receive the path... and the second is because our
implementation require the file to be allowed to be mapped, so we can get a
name... so i added the debug to see if there's any case where we need to
improve this function... (basically a total extra complex rewrite :P)

Em ter, 2 de out de 2018 às 22:43, Svyatpro <notifications@github.com>
escreveu:

**Comment by pachuco** [comment]
Perhaps forward to OutputDebugString instead of MessageBox :p

**Comment by pachuco** [comment]
The function remains the same, but I've replaced the debug portion for less intrusiveness. Yay!
